src/main/java/com/example/onlinebookstore/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,22 @@
+package com.example.onlinebookstore.security;
+
+import com.example.onlinebookstore.exceptions.EntityNotFoundException;
+import com.example.onlinebookstore.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findUserByEmail(username)
+                .orElseThrow(() -> new EntityNotFoundException("No such user with this email: "
```

Comment:
```suggestion
                .orElseThrow(() -> new UsernameNotFoundException("No such user with this email: "
```

and remove UsernameNotFoundException from signature